pkg/csi/cinder/openstack/openstack_volumes.go

```diff
@@ -186,14 +187,23 @@ func (os *OpenStack) AttachVolume(instanceID, volumeID string) (string, error) {
 		computeServiceClient.Microversion = "2.60"
 	}
 
-	_, err = volumeattach.Create(computeServiceClient, instanceID, &volumeattach.CreateOpts{
+	vAttachRes, err := volumeattach.Create(computeServiceClient, instanceID, &volumeattach.CreateOpts{
 		VolumeID: volume.ID,
 	}).Extract()
 
 	if err != nil {
 		return "", fmt.Errorf("failed to attach %s volume to %s compute: %v", volumeID, instanceID, err)
 	}
 
+	if readOnly {
```

Comment:
not sure why we can't add this param when create volume attach at l 190?

Reply:
@jichenjc Its not supported

Reply:
it's supported in cinder
https://docs.openstack.org/api-ref/block-storage/v3/?expanded=update-an-attachment-detail,create-attachment-detail#create-attachment

so maybe a follow up PR can go to gophercloud to add this param for create?

Reply:
https://github.com/gophercloud/gophercloud/blob/master/openstack/blockstorage/v3/attachments/requests.go#L31

looks like it support ro/rw already?

Reply:
@jichenjc we are using call compute API call for volume attach . Pls check L190. ITs not supported there .
And also , its not recommended to use blockstorage api directly for volume attach as you mentioned for volume attach (that should be internally called by nova), because that misses the update in nova DB . So we need to add additional call for updating the mode. Hope its clear now.

Reply:
ok, I need check compute calls and see whether we can add some params
IMHO it should create with param instead of update later on..

Reply:
ok, read the code again and seems we are using nova api to create the attachment 
and use cinder api to update the attachment..  not sure this is desired way but this looks like what it is
and no API of nova side to specify the ro now